tests/Codec/LazyBSONDocumentCodecTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace MongoDB\Tests\Codec;
+
+use MongoDB\BSON\Document;
+use MongoDB\Codec\LazyBSONDocumentCodec;
+use MongoDB\Exception\UnsupportedValueException;
+use MongoDB\Model\LazyBSONDocument;
+use MongoDB\Tests\TestCase;
+
+class LazyBSONDocumentCodecTest extends TestCase
+{
+    private const OBJECT = [
+        'foo' => 'bar',
+        'document' => ['bar' => 'baz'],
+        'array' => [0, 1, 2],
+    ];
+
+    public function testDecode(): void
+    {
+        $document = (new LazyBSONDocumentCodec())->decode($this->getTestDocument());
+
+        $this->assertInstanceOf(LazyBSONDocument::class, $document);
+        $this->assertSame('bar', $document->foo);
+    }
+
+    public function testDecodeWithWrongType(): void
+    {
+        $codec = new LazyBSONDocumentCodec();
+
+        $this->expectExceptionObject(UnsupportedValueException::invalidDecodableValue('foo'));
+        $codec->decode('foo');
+    }
+
+    public function testEncode(): void
+    {
+        $document = new LazyBSONDocument($this->getTestDocument());
+        $encoded = (new LazyBSONDocumentCodec())->encode($document);
+
+        $this->assertEquals(
+            self::OBJECT,
+            $encoded->toPHP(['root' => 'array', 'array' => 'array', 'document' => 'array']),
+        );
```

Comment:
I didn't think to ask this in LazyBSONArrayCodecTest, but why do you rely on `toPHP()` here isntead of comparing Document objects directly? Or arrays in the case of LazyBSONArrayCodecTest's corresponding method.

Reply:
That's a good point. I've changed both assertions to compare the encoded with a `Document` instance (and `PackedArray` in `LazyBSONArrayCodecTest`).